wasmcloud_host/lib/wasmcloud_host_web/controllers/provider_controller.ex

```diff
@@ -0,0 +1,30 @@
+defmodule WasmcloudHostWeb.ProviderController do
+  use WasmcloudHostWeb, :controller
+  require HostCore
+
+  def start_provider(conn, params) do
+    upload = params["provider_file"]
+    key = params["provider_key"]
+    contract_id = params["provider_contract_id"]
+    link_name = params["provider_link_name"]
+
+    # Temporary logic to write provider to temp dir.
```

Comment:
Let's create an issue that comes after the conversion to supporting OCI for providers and actors that bans the use of uploaded executables and can only handle uploaded PARs or OCI-downloaded PARs.

Reply:
Wrote up https://github.com/wasmCloud/wasmcloud-otp/issues/38 for this issue